common/network-shuffle/src/main/java/org/apache/spark/network/shuffle/protocol/StreamHandle.java

```diff
@@ -57,25 +67,31 @@ public boolean equals(Object other) {
     if (other != null && other instanceof StreamHandle) {
       StreamHandle o = (StreamHandle) other;
       return Objects.equal(streamId, o.streamId)
-        && Objects.equal(numChunks, o.numChunks);
+        && Objects.equal(numChunks, o.numChunks)
+        && Arrays.equals(chunkSizes, o.chunkSizes);
     }
     return false;
   }
 
   @Override
   public int encodedLength() {
-    return 8 + 4;
+    return 8 + 4 + Encoders.IntArrays.encodedLength(chunkSizes);
   }
 
   @Override
   public void encode(ByteBuf buf) {
     buf.writeLong(streamId);
     buf.writeInt(numChunks);
+    Encoders.IntArrays.encode(buf, chunkSizes);
   }
 
   public static StreamHandle decode(ByteBuf buf) {
     long streamId = buf.readLong();
     int numChunks = buf.readInt();
-    return new StreamHandle(streamId, numChunks);
+    int[] chunkSizes = new int[0];
```

Comment:
we should avoid creating the 0-size array again and again.

Reply:
Got it, many place in this PR has this issue, I will fix them.